frontend/src/metabase-lib/native.ts

```diff
@@ -4,6 +4,53 @@ import type { DatabaseId } from "metabase-types/api";
 
 import type { MetadataProvider, Query } from "./types";
 
+export function nativeQuery(
+  databaseId: DatabaseId,
+  metadata: MetadataProvider,
+  innerQuery: string,
+): Query {
+  return ML.native_query(databaseId, metadata, innerQuery);
+}
+
+export function rawNativeQuery(query: Query): string {
+  return ML.raw_native_query(query);
+}
+
+export function withNativeQuery(query: Query, innerQuery: string): Query {
+  return ML.with_native_query(query, innerQuery);
+}
+
+export function withTemplateTags(query: Query, tags: string[]): Query {
+  return ML.with_template_tags(query, tags);
+}
+
+export function templateTags(query: Query): string {
```

Comment:
Perhaps `TemplateTag[]` or `Record<string, TemplateTag>`? Check what the function actually returns